src/checkoutservice/main.go

```diff
@@ -428,3 +430,16 @@ func (cs *checkoutService) shipOrder(ctx context.Context, address *pb.Address, i
 }
 
 // TODO: Dial and create client once, reuse.
+
+func getTraceLogFields(ctx context.Context) logrus.Fields {
+	span := trace.FromContext(ctx)
+	if span == nil {
+		return logrus.Fields{}
+	}
+	traceID := span.SpanContext().TraceID
+	spanID := span.SpanContext().SpanID
+	return logrus.Fields{
+		"trace_id": hex.EncodeToString(traceID[:]),
+		"span_id":  hex.EncodeToString(spanID[:]),
+	}
```

Comment:
If it is easy it would be nice to extract this portion as a function and reuse. This part will possible change in the future (e.g. trace_id/span_end encoding may change from hex to something else), so having it in one place will make changes easier.

Reply:
I can do it in a future PR. need to merge this one before https://github.com/signalfx/microservices-demo/pull/8